azurerm/resource_arm_api_management_group.go

```diff
@@ -0,0 +1,165 @@
+)
+
+func resourceArmAPIManagementGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAPIManagementGroupCreateUpdate,
+		Read:   resourceArmAPIManagementGroupRead,
+		Update: resourceArmAPIManagementGroupCreateUpdate,
+		Delete: resourceArmAPIManagementGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"api_management_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"display_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"external_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(apimanagement.Custom),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(apimanagement.Custom),
+					string(apimanagement.External),
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceArmAPIManagementGroupCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).apiManagementGroupClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	externalID := d.Get("external_id").(string)
+	groupType := d.Get("type").(string)
+
+	parameters := apimanagement.GroupCreateParameters{
+		GroupCreateParametersProperties: &apimanagement.GroupCreateParametersProperties{
+			DisplayName: utils.String(displayName),
+			Description: utils.String(description),
+			ExternalID:  utils.String(externalID),
+			Type:        apimanagement.GroupType(groupType),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, name, parameters, ""); err != nil {
+		return fmt.Errorf("Error creating or updating API Management Group %q (resource group %q, API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, name)
+	if err != nil {
+		return fmt.Errorf("Error getting API Management Group %q (resource group %q, API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read API Management Group %q (resource group %q, API Management Service %q) ID", name, resourceGroup, serviceName)
```

Comment:
_minor_ in the other resources we use:

```suggestion
		return fmt.Errorf("Cannot read Group %q (Resource Group %q / API Management Service %q) ID", name, resourceGroup, serviceName)
```